File: app/src/main/java/com/adsamcik/signalcollector/receivers/NotificationReceiver.java
package com.adsamcik.signalcollector.receivers;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;

import com.adsamcik.signalcollector.services.TrackerService;
import com.adsamcik.signalcollector.utility.FirebaseAssist;
import com.adsamcik.signalcollector.utility.Preferences;
import com.google.firebase.analytics.FirebaseAnalytics;

public class NotificationReceiver extends BroadcastReceiver {
	private final String TAG = "SignalsNotifiReceiver";
	public static final String ACTION_STRING = "action";

	@Override
	public void onReceive(Context context, Intent intent) {
		int value = intent.getIntExtra(ACTION_STRING, -1);
		Bundle params = new Bundle();
		params.putString(FirebaseAssist.PARAM_SOURCE, "notification");
		switch (value) {
			case 0:
				Preferences.stopTillRecharge(context);
				FirebaseAnalytics.getInstance(context).logEvent(FirebaseAssist.STOP_TILL_RECHARGE_EVENT, params);
				break;
			case 1:
				FirebaseAnalytics.getInstance(context).logEvent(FirebaseAssist.STOP_EVENT, params);
				context.stopService(new Intent(context, TrackerService.class));
				break;
			default:
				Log.w(TAG, "Unknown value " + value);
		}
	}
}

File: app/src/main/java/com/adsamcik/signalcollector/receivers/NotificationReceiver.kt
package com.adsamcik.signalcollector.receivers


import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import com.adsamcik.signalcollector.services.TrackerService
import com.adsamcik.signalcollector.utility.Constants
import com.adsamcik.signalcollector.utility.FirebaseAssist
import com.adsamcik.signalcollector.utility.TrackingLocker
import com.google.firebase.analytics.FirebaseAnalytics

class NotificationReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val value = intent.getIntExtra(ACTION_STRING, -1)
        val params = Bundle()
        params.putString(FirebaseAssist.PARAM_SOURCE, "notification")
        when (value) {
            0 -> {
                TrackingLocker.lockUntilRecharge(context)
                FirebaseAnalytics.getInstance(context).logEvent(FirebaseAssist.STOP_TILL_RECHARGE_EVENT, params)
            }
            1 -> {
                FirebaseAnalytics.getInstance(context).logEvent(FirebaseAssist.STOP_EVENT, params)
                context.stopService(Intent(context, TrackerService::class.java))
            }
            2 -> {
                val minutes = intent.getIntExtra(STOP_MINUTES_EXTRA, -1)
                if (minutes > 0)
                    TrackingLocker.lockTimeLock(context, Constants.MINUTE_IN_MILLISECONDS * minutes)
            }
            else -> Log.w(TAG, "Unknown value $value")
        }
    }

    companion object {
        private const val TAG = "SignalsReceiver"
        const val ACTION_STRING = "action"
        const val STOP_MINUTES_EXTRA = "stopForMinutes"
    }
}
